### PR TITLE
Save metadata in separate metadata map

### PR DESCRIPTION
**Pull request by timbussmann**
Adds some additional metadata information about the saga data into a special metadata map attribute.
The saga version attribute used for MVCC checks is also moved into the metadata to cause less "pollution" of the saga data root-level attributes.

The pessimistic locking lease attribute remains at the root level though as it is exceptionally hard (if not impossible) to upsert a map with an element or adding the element otherwise in a single operation. Ultimately this is fine as the lease technically belongs to the saga data but is just stored on the same ID for optimization reasons.

### CODE REVIEW

File: src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs [hunk 1]
```diff
@@ -255,11 +261,17 @@ public Task Update(IContainSagaData sagaData, ISynchronizedStorageSession sessio
...
             map.Add(configuration.Table.SortKeyName, new AttributeValue { S = $"SAGA#{sagaData.Id}" });  //Sort key
-            // Version should probably be properly moved into metadata to not clash with existing things
-            map.Add(SagaDataVersionAttributeName, new AttributeValue { N = version.ToString() });
+            map.Add(SagaMetadataAttributeName, new AttributeValue
+            {
+                M = new Dictionary<string, AttributeValue>()
+                {
+                    { SagaDataVersionAttributeName, new AttributeValue { N = version.ToString() } },
+                    { "TYPE", new AttributeValue { S = sagaData.GetType().FullName } },
+                    { "SCHEMA_VERSION", new AttributeValue { N = "1"} }
```
Comment (timbussmann) [inline src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs:270]: we could also use a string to contain a MinVer number instead but I assumed that a single number will be enough for any internal handling of different schema versions.

**Review by kentdr — APPROVED** [review]

File: src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs (hunk 1 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs:270]: I would prefer here at least 1.0 or 1.0.0

**Review by danielmarbach — APPROVED** [review]
Do we need the same metadata also for the outbox records? In other persisters we did that as far as I recall

File: src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs (hunk 1 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs:270]: Can be addressed later

**Comment by timbussmann** [comment]
> Do we need the same metadata also for the outbox records? In other persisters we did that as far as I recall

I quickly looked at it but I don't think it's necessary since we have a dedicated metadata record that we can freely use and there is no "saga type" information. So the only thing we would track is the version which we could add retroactively fairly easy?

**Pull request merged**